### ISSUE TITLE
Trying to use kotlinx-datetime

### ISSUE THREAD

**Issue by ivan-brko** [Litote/kmongo#297]
Hi, 

I'm trying to use `kotlinx-datetime` library for dates with KMongo (with `kotlinx-serialization`) and as I am new to the library I'm trying to figure our how I should register serializers/deserializers for `kotlinx-datetime` types with KMongo.

From what I've seen in the library, if I did the following should that work: 
1) Implement `TemporalExtendedJsonSerializer<T>` for some type from `kotlinx-datetime`
2) Register it with `registerSerializer()`

Is this correct? And even if it is, is there a better way to handle this? Also, I know that `kotlinx-datetime` is still in early phase, but is there a plan to support it in the future?

**Comment by ivan-brko** [comment]
The procedure I gave above seems to work, but the resulting field is stored as String in Mongo, not as Date.

**Comment by ivan-brko** [comment]
Figured out the issue, so writing it down for anyone else who wants to use `kotlinx-datetime`.
I guess the issue is I don't know enough about which serializers/deserializers are used by kotlinx-serialization/KMongo. 

Even though I registered `TemporalExtendedJsonSerializer<T>` with `registerSerializer()` the code was in the end still using default String serialization for these types provided by `kotlinx-datetime` library. 
The solution for my data classes which are used with KMongo was to annotate the fields of these types with     `@Serializable(with = MyLocalDateSerializer::class)`
(where `MyLocalDateSerializer` is a serializer I wrote which extends `TemporalExtendedJsonSerializer<T>`).
